dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java

```diff
@@ -0,0 +1,308 @@
+package com.dotcms.storage;
+
+import com.dotcms.tika.TikaUtils;
+import com.dotcms.util.MimeTypeUtils;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.FileUtil;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableSortedMap;
+import io.vavr.control.Try;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+/**
+ * Default implementation
+ * @author jsanca
+ */
+public class FileStorageAPIImpl implements FileStorageAPI {
+
+    private static final String CACHE_GROUP = "Contentlet";
+
+    // width,height,contentType,author,keywords,fileSize,content,length,title
+
+    private volatile ObjectReaderDelegate objectReaderDelegate = new JsonReaderDelegate<>(Map.class);
+    private volatile ObjectWriterDelegate objectWriterDelegate = new JsonWriterDelegate();
+    private volatile MetadataGenerator    metadataGenerator    = new TikaMetadataGenerator();
+    private final    StorageProvider      storageProvider      = new StorageProvider();
+
+    @Override
+    public void setObjectReaderDelegate(final ObjectReaderDelegate objectReaderDelegate) {
+        this.objectReaderDelegate = objectReaderDelegate;
+    }
+
+    @Override
+    public void setObjectWriterDelegate(final ObjectWriterDelegate objectWriterDelegate) {
+        this.objectWriterDelegate = objectWriterDelegate;
+    }
+
+    @Override
+    public void setMetadataGenerator(final MetadataGenerator metadataGenerator) {
+        this.metadataGenerator = metadataGenerator;
+    }
+
+    public StorageProvider getStorageProvider() {
+        return this.storageProvider;
+    }
+
+    @Override
+    public Map<String, Object> generateRawBasicMetaData(final File binary) {
+
+        return this.generateBasicMetaData(binary, s -> true); // raw = no filter
+    }
+
+    @Override
+    public Map<String, Object> generateRawFullMetaData(final  File binary, long maxLength) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'maxLength' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)